basyx/aas/adapter/http.py

```diff
@@ -228,9 +228,12 @@ class HTTPApiDecoder:
         model.SpecificAssetId: XMLConstructables.SPECIFIC_ASSET_ID,
         model.Qualifier: XMLConstructables.QUALIFIER,
         model.Submodel: XMLConstructables.SUBMODEL,
-        model.SubmodelElement: XMLConstructables.SUBMODEL_ELEMENT
+        model.SubmodelElement: XMLConstructables.SUBMODEL_ELEMENT,
+        model.Reference: XMLConstructables.REFERENCE
     }
 
+    encoding = "utf-8"
```

Comment:
The `encoding` should be consistent for encoding and decoding. Since this is the class HTTPApi**Decoder**, we should instead move the `encoding` constant and the respective functions to the module scope or to a separate class.